File: estate_app/estate_plan/web_form/student_contact/student_contact.py
# import frappe

# def get_context(context):
# 	# do your magic here
# 	pass
# In your custom app (e.g., estate_plan)

import frappe
from frappe import _

@frappe.whitelist()
def send_email_from_form(doc, subject, message):
    try:
        # Construct email parameters
        recipients = doc.get('email')
        
        # Send email using frappe's email utility
        frappe.sendmail(recipients=recipients, subject=subject, message=message)
        
        return 'success'
    except Exception as e:
        frappe.log_error(_('Failed to send email: {0}').format(str(e)))
        return 'failed'

def get_context(context):
    # Initialize an empty context dictionary
    context = {}

    # Populate the context dictionary with necessary data
    # Example:
    context['title'] = 'Student Contact Details'

    # Return the populated context dictionary
    return context

File: estate_app/estate_plan/web_form/student_contact/student_contact.js
// frappe.ready(function() {
//     // Bind events here if needed


// frappe.ui.form.on('Student Contact Details', {
//     // Triggered after the form is saved successfully
//     onload_post_render: function(frm) {
//         frm.add_custom_button('Send Email', function() {
//             sendEmail(frm);
//         });
//     }
// });

// function sendEmail(frm) {
//     // Replace with your email subject and content
//     var subject = 'New Student Contact Details';
//     var message = 'A new student contact details record has been created:\n\n' +
//                   'Name: ' + frm.doc.student_name + '\n' +
//                   'Email: ' + frm.doc.email + '\n';

//     frappe.call({
//         method: 'send_email_from_form', // Replace with your server-side method
//         args: {
//             doc: frm.doc,
//             subject: subject,
//             message: message
//         },
//         callback: function(response) {
//             if (response && response.message === 'success') {
//                 frappe.msgprint('Email sent successfully.');
//             } else {
//                 frappe.msgprint('Failed to send email.');
//             }
//         }
//     });
// }
// });


